app/models/debate.rb

```diff
@@ -11,23 +11,30 @@ class Debate < ApplicationRecord
 
   accepts_nested_attributes_for :answers
   validates :topic, presence: true
-  validates :closed_at, presence: true
   validates :code, presence: true, uniqueness: true,
                    length: { is: CODE_LENGTH }, numericality: { only_integer: true }
 
   before_save  :block_code_change
 
-  attribute :closed_at, :datetime, default: -> { Time.current + 1.hour }
+  attribute :is_closed, :boolean, default: false
+  alias_attribute :closed?, :is_closed
+
+  mattr_accessor :debate_notifier
+  self.debate_notifier = DebateNotifier.build
 
   def self.opened_for_channel!(channel_name)
     where(
-      "channel_name = ? AND closed_at > ?",
-       channel_name, Time.current
+      "channel_name = ? AND is_closed = 'f'",
+       channel_name
     ).first!
```

Comment:
It's possible to use active record query:

`where(channel_name: channel_name, is_closed: false)`

Also, it might be good to define a named scope for closed debates.